pandas/plotting/_core.py

```diff
@@ -625,11 +626,43 @@ def _get_plot_backend():
     The backend is imported lazily, as matplotlib is a soft dependency, and
     pandas can be used without it being installed.
     """
-    try:
-        import pandas.plotting._matplotlib as plot_backend
-    except ImportError:
-        raise ImportError("matplotlib is required for plotting.")
-    return plot_backend
+    backend_str = pandas.get_option('plotting.backend')
+    if backend_str == 'matplotlib':
+        try:
+            import pandas.plotting._matplotlib as backend_mod
+        except ImportError:
+            raise ImportError('matplotlib is required for plotting when the '
+                              'default backend is selected.')
+    else:
+        try:
+            mod = importlib.import_module(backend_str)
+        except ImportError:
+            raise ValueError('"{}" does not seem to be an installed module.'
+                             'A pandas plotting backend must be a module that '
+                             'can be imported'.format(backend_str))
+
+        required_objs = ['LinePlot', 'BarPlot', 'BarhPlot', 'HistPlot',
```

Comment:
This seems like we’re being too opinionated about the backends implementation. IMO, we should provide a series and frame .plot accessor that dispatches to the right backend. We can provide an ABC for backends to subclass with the expected user API, but beyond that I don’t think we care.

Reply:
I don't have a strong opinion about this at this point. I opened #26747 to have a discussion on what we expect from backends, this can surely be simplified.

At this stage I just tried to be conservative, and force the backends to raise `NotImplementedError` or whatever they consider if they don't implement something. My idea here is that if the user calls something like `Series.plot()` or any other functionality, we don't find with a weird error that doesn't provide relevant information for the user.

But happy raise exceptions from our side for some of those when the backend is not the default one, if that's the preferred option.